### PR TITLE
Fix/tao 4351 skip authorization

### PR DESCRIPTION
**Pull request by jsconan**
Related to: https://oat-sa.atlassian.net/browse/TAO-4351

Requires: https://github.com/oat-sa/extension-tao-ltideliveryprovider/pull/85

Pause running deliveries when launching from LTI.

### CODE REVIEW

File: manifest.php [hunk 1]
```diff
@@ -15,18 +15,15 @@
...
-use oat\ltiProctoring\controller\ProctoringTool;
+use oat\ltiProctoring\controller\DeliveryServer;
 use oat\ltiProctoring\controller\Monitor;
+use oat\ltiProctoring\controller\ProctoringTool;
 use oat\ltiProctoring\controller\Reporting;
-use oat\taoLti\models\classes\LtiRoles;
-use oat\tao\model\user\TaoRoles;
-use oat\ltiProctoring\scripts\install\SetupProctoringEventListeners;
-use oat\ltiProctoring\scripts\install\RegisterAuthProvider;
 use oat\ltiProctoring\scripts\install\RegisterServices;
```
Comment (hutnikau) [inline manifest.php]: Please do not do formatting and other changes which are not related to fix. It makes review much harder. Do that it separate pull request.

File: model/LtiListenerService.php [hunk 2]
```diff
@@ -125,6 +126,27 @@ public function executionStateChanged(DeliveryExecutionState $event)
...
+        $user = $event->getUser();
+        $remoteLink = \taoLti_models_classes_LtiService::singleton()->getLtiSession()->getLtiLinkResource();
+        $executions = LTIDeliveryTool::singleton()->getLinkedDeliveryExecutions($delivery, $remoteLink, $user->getIdentifier());
+        if (!empty($executions)) {
+            $deliveryExecutionService = $this->getServiceManager()->get(LtiDeliveryExecutionService::SERVICE_ID);
+            foreach ($executions as $deliveryExecution) {
+                if (!$deliveryExecutionService->isFinished($deliveryExecution) && $deliveryExecution->getState()->getUri() != DeliveryExecution::STATE_PAUSED) {
+                    /** @var DeliveryExecutionStateService $deliveryExecutionStateService */
+                    $deliveryExecutionStateService = $this->getServiceManager()->get(DeliveryExecutionStateService::SERVICE_ID);
+                    $deliveryExecutionStateService->pauseExecution($deliveryExecution);
```
Comment (hutnikau) [inline model/LtiListenerService.php]: I guess this solution is not optimal.
All the active executions will be paused. This may be not safe.
Also i don't like that this is event based functionality. This may be not reliable.

Please look at this draft:
https://github.com/oat-sa/extension-tao-ltideliveryprovider/compare/develop...fix/pause_authorized_execution_during_lti_launch?expand=1

You may add checking of state before pausing there. Also this approach requires changes in only one extension.

Comment (hutnikau) [inline model/LtiListenerService.php]: Also please consider to pass pause reason as the second parameter of `pauseExecution()`

**Comment by jsconan** [comment]
Ok, I applied your solution in the main PR.
The use of event was to avoid to introduce proctoring stuff in the regular LTI mode.
I tested your code with and without proctoring mode, to check for issues, and I made a little adjustment to make it compatible.
